compiler/rustc_trait_selection/src/traits/object_safety.rs

```diff
@@ -157,6 +158,11 @@ fn object_safety_violations_for_trait(
                 .in_definition_order()
                 .filter(|item| item.kind == ty::AssocKind::Type)
                 .filter(|item| !tcx.generics_of(item.def_id).params.is_empty())
+                .filter(|item| {
+                    tcx.def_path(item.def_id).data.last().map_or(true, |def_path_data| {
+                        def_path_data.data != DefPathData::ImplTraitAssocTy
+                    })
+                })
```

Comment:
RPITITs are not object safe. Is this to prevent an ICE or what? It should probably turn into a new `ObjectSafetyViolation` kind.

Reply:
Removed, it will be easier to justify with tests later.